sklearn/ensemble/forest.py

```diff
@@ -354,6 +384,17 @@ def _validate_X_predict(self, X):
 
         return self.estimators_[0]._validate_X_predict(X, check_input=True)
 
+    def _validate_missing_mask(self, X, missing_mask=None):
```

Comment:
I am wondering whether adding this `missing_mask` really is bringing anything? Wouldnt it have been the same to compare directly the feature values when splitting and predicting? This would have saved a lot of boilerplate code to transport this array from one place to another. What do you think? Maybe I am missing something,